api/internal/core/storage/adminapi.go

```diff
@@ -0,0 +1,190 @@
+
+	"github.com/apache/apisix-dashboard/api/pkg/storage"
+)
+
+const (
+	adminAPIErrorWarp = "failed to request Admin API"
+)
+
+type adminAPIVersion = int
+
+const (
+	adminAPIVersion2 adminAPIVersion = 2
+	adminApiVersion3 adminAPIVersion = 3
+)
+
+// Ensure that the AdminAPI Storage implementation conforms to the storage_api interface definition
+var _ storage_api.Interface = AdminAPIStorage{}
+
+// NewAdminAPIStorage will create an instance of AdminAPIStorage and complete the basic configure
+func NewAdminAPIStorage() *AdminAPIStorage {
+	return &AdminAPIStorage{
+		client: resty.New().
+			SetBaseURL("http://127.0.0.1:9080/apisix/admin/").
+			SetHeader("X-API-KEY", "edd1c9f034335f136f87ad84b625c8f1").
+			SetTimeout(1 * time.Second),
+	}
+}
+
+// AdminAPIStorage is the storage layer implementation of the APISIX Admin API
+type AdminAPIStorage struct {
+	client *resty.Client
+}
+
+func (s AdminAPIStorage) Get(ctx context.Context, key string) (string, error) {
+	resp, err := s.client.R().
+		SetContext(ctx).
+		Get(s.removeKeyPrefix(key))
+	if err != nil {
+		return "", errors.Wrap(err, adminAPIErrorWarp)
+	}
+
+	json := gjson.Parse(resp.String())
+	if json.Get("count").Int() < 1 {
+		return "", errors.Errorf("key: %s is not found", key)
+	}
+
+	return gjson.Get(resp.String(), "node.value").Raw, nil
+}
+
+func (s AdminAPIStorage) List(ctx context.Context, key string) ([]storage_api.Keypair, error) {
+	resp, err := s.client.R().
+		SetContext(ctx).
+		Get(s.removeKeyPrefix(key))
+	if err != nil {
+		return nil, errors.Wrap(err, adminAPIErrorWarp)
+	}
+
+	switch resp.Header().Get("X-API-VERSION") {
+	case "v3":
+		return s.parseList(adminApiVersion3, resp.String()), nil
+	case "v2":
+		fallthrough
+	default:
+		return s.parseList(adminAPIVersion2, resp.String()), nil
+	}
+}
+
+func (s AdminAPIStorage) Create(ctx context.Context, key, val string) error {
+	return s.Update(ctx, key, val)
+}
+
+func (s AdminAPIStorage) Update(ctx context.Context, key, val string) error {
+	resp, err := s.client.R().
+		SetContext(ctx).
+		SetHeader("Content-Type", "application/json").
+		SetBody(val).
+		Put(s.removeKeyPrefix(key))
+	if err != nil {
+		return errors.Wrap(err, adminAPIErrorWarp)
+	}
+
+	if err = s.checkError(resp.String()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s AdminAPIStorage) BatchDelete(ctx context.Context, keys []string) error {
+	for _, key := range keys {
+		resp, err := s.client.R().
+			SetContext(ctx).
+			Delete(s.removeKeyPrefix(key))
+		if err != nil {
+			return errors.Wrap(err, adminAPIErrorWarp)
+		}
+
+		if err = s.checkError(resp.String()); err != nil {
+			return fmt.Errorf("delete key[%s] failed: %s", key, err)
+		}
+
+		if gjson.Get(resp.String(), "deleted").Int() < 1 {
+			return errors.Errorf("key: %s is not found", key)
+		}
+	}
+
+	return nil
+}
+
+// Watch watches for changes in data sources and generates event streams
+// Admin API storage implementation does not support Watch capability
```

Comment:
So how should we synchronize incremental changes? Do we have a proposal for admin api storage?

Reply:
This `Watch` is used to update the cache in the `GenericStore`, and I will add a switch to it to allow it to turn off the cache to fetch data directly from the data source rather than through the Watch's caching mechanism.

Reply:
Or even if we don't continue to use Etcd directly as a data source, we can just remove it.

Reply:
Here I only leave an empty implementation, if you must implement it will be more complicated.

Use timers to fetch data in full at regular intervals to update the cache, and when an operation to update the data is triggered, the data is updated in full, compared with the value in memory, and converted to a change event to be returned to the upper layer.

Reply:
> if you must implement it will be more complicated.

no. just make sure you have a plan for it.